NIPostsReload/Presentation/Scenes/NIPostFeedModule/ViewState/NIPostViewState.swift

```diff
@@ -9,37 +9,54 @@ import Foundation
 
 struct NIPostViewState {
     struct Post {
-        let postId: Int
         let title: String
         let previewText: String
         let date: String
-        let likesCount: Int
+        let likesCount: String
         
         var isExpanded = false
     }
     
-    var posts: [Post]
-}
-
-extension NIPostViewState {
-    static func makeViewState(for posts: [NIPost]) -> NIPostViewState {
+    var items: [Post] = []
+    private var posts: [NIPost] = []
+    
+    mutating func sort(by sortType: PostFeedSortType) {
+        switch sortType {
+        case .date:
+            posts.sort { $0.timeshamp ?? 0 > $1.timeshamp ?? 0 }
+        case .popularity:
+            posts.sort { $0.likesCount ?? 0 > $1.likesCount ?? 0}
+        case .default:
+            posts.sort { $0.postId < $1.postId }
+        }
+        
+        makeViewState()
+    }
+    
+    func getPost(ad index: Int) -> NIPost {
```

Comment:
I recommend separating the functions that involve muting from those that do not involve muting